### PR TITLE
Refactor worker result approve to worker payment approve #1251

### PR DESCRIPTION
**Pull request by maslenitsa93**
#1251

### CODE REVIEW

File: libraries/chain/worker_evaluators.cpp [hunk 1]
```diff
@@ -322,22 +322,22 @@ namespace golos { namespace chain {
...
-        const auto& wto = _db.get_worker_result(worker_result_post.id);
+        const auto& wto_post = _db.get_comment(o.worker_techspec_author, o.worker_techspec_permlink);
+        const auto& wto = _db.get_worker_techspec(wto_post.id);
 
         const auto& wpo = _db.get_worker_proposal(wto.worker_proposal_post);
 
         GOLOS_CHECK_LOGIC(wto.state == worker_techspec_state::complete || wto.state == worker_techspec_state::payment,
-            logic_exception::worker_techspec_should_be_complete_or_paying_to_approve_result,
-            "Worker techspec should be complete or paying to approve result");
+            logic_exception::worker_techspec_should_be_complete_or_paying,
```
Comment (afalaleev) [inline libraries/chain/worker_evaluators.cpp]: `worker_techspec_state::work` ?
`worker_techspec_state::wip` ?

File: libraries/chain/worker_evaluators.cpp [hunk 2]
```diff
@@ -346,37 +346,38 @@ namespace golos { namespace chain {
...
 
-        if (wrao_itr != wrao_idx.end()) {
-            WORKER_CHECK_NO_VOTE_REPEAT(wrao_itr->state, o.state);
+        if (wpao_itr != wpao_idx.end()) {
+            WORKER_CHECK_NO_VOTE_REPEAT(wpao_itr->state, o.state);
 
-            _db.modify(*wrao_itr, [&](worker_result_approve_object& wrao) {
-                wrao.state = o.state;
+            _db.modify(*wpao_itr, [&](worker_payment_approve_object& wpao) {
+                wpao.state = o.state;
```
Comment (afalaleev) [inline libraries/chain/worker_evaluators.cpp]: Approves available only on `worker_techspec_state::complete`
Disapproves available on all states after `worker_techspec_state::approved`

File: libraries/chain/worker_evaluators.cpp [hunk 3]
```diff
@@ -346,37 +346,38 @@ namespace golos { namespace chain {
...
             });
         } else {
-            _db.create<worker_result_approve_object>([&](worker_result_approve_object& wrao) {
-                wrao.approver = o.approver;
-                wrao.post = worker_result_post.id;
-                wrao.state = o.state;
+            _db.create<worker_payment_approve_object>([&](worker_payment_approve_object& wpao) {
+                wpao.approver = o.approver;
+                wpao.post = wto_post.id;
+                wpao.state = o.state;
```
Comment (afalaleev) [inline libraries/chain/worker_evaluators.cpp]: The same as previous comment.

File: libraries/chain/worker_evaluators.cpp [hunk 4]
```diff
@@ -346,37 +346,38 @@ namespace golos { namespace chain {
...
             });
         }
 
-        auto approves = _db.count_worker_result_approves(worker_result_post.id);
+        auto approves = _db.count_worker_payment_approves(wto_post.id);
 
         if (o.state == worker_techspec_approve_state::disapprove) {
```
Comment (afalaleev) [inline libraries/chain/worker_evaluators.cpp]: On disapproving the `_db.close_worker_techspec()` should be called

File: libraries/chain/worker_evaluators.cpp (hunk 1 above)
Comment (maslenitsa93) [reply libraries/chain/worker_evaluators.cpp]: @afalaleev In another pull request.

**Comment by maslenitsa93** [comment]
Updating metadata on closing techspec by author should be tested in https://github.com/GolosChain/golos/issues/1227

**Pull request closed without merging**